docs/Using-SRIOV-and-Clear-Containers.md

```diff
@@ -0,0 +1,265 @@
+   you may resolve this by adding a `pcie_acs_override=` option to your kernel 
+   command line and reboot. Detailed information about this option can be found 
+   at [PCIE-ACS-override-option](https://lkml.org/lkml/2013/5/30/513).
+
+## Setting up SR-IOV Device
+
+All the steps in prior sections need to be performed just once to prepare the 
+SR-IOV host systems. The following will be needed per system boot in order to
+facilitate setting up a physical device's virtual functions.
+
+The `sriov_totalvfs` file under `sysfs` for a pci device specifies the total
+number of VFs that can be created. VFs can be created by editing the file `sriov_numvfs`.
+Once VF's are created, you cannot grow or shrink the number of VFs without 
+first setting the number back to zero. Based on this, it is expected that you
+should set the number of VFs for a physical device just once.
+
+1. Add `vfio-pci` device driver
+
+   `vfio-pci` is a driver which is used to reserve a VF PCI device.  Add it:
+   ```
+   $ sudo modprobe vfio-pci
+   ```
+
+2. Find our NICs of interest
+
+   Find PCI details for the NICs in question:
+   ```
+   $ lspci | grep Ethernet
+   00:19.0 Ethernet controller: Intel Corporation Ethernet Connection I217-LM (rev 04)
+   01:00.0 Ethernet controller: Intel Corporation Ethernet Controller 10-Gigabit X540-AT2 (rev 01)
+   01:00.1 Ethernet controller: Intel Corporation Ethernet Controller 10-Gigabit X540-AT2 (rev 01)
+   ```
+   In our case, both 01:00.0 and 01:00.1 are the two ports on our x540-AT2 card 
+   that we'll use.  You can use `lshw` command to get further details on the 
+   controller and verify it indeed supports SR-IOV.
+
+3. Check how many VFs we can create:
+
+   ```
+   $ cat /sys/bus/pci/devices/0000\:01\:00.0/sriov_totalvfs
+   63
+   $ cat /sys/bus/pci/devices/0000\:01\:00.1/sriov_totalvfs
+   63
+   ```
+
+4. Create VFs:
+
+   Create virtual functions by editing `sriov_numvfs`. In our example, we create
+   one per physical device.  Note, this eliminates the usefulness 
+   of SR-IOV, and is just done for simplicity in this example so you needn't
+   look at 128 virtual devices.
+
+   ```
+   # echo 1 > /sys/bus/pci/devices/0000\:01\:00.0/sriov_numvfs
+   # echo 1 > /sys/bus/pci/devices/0000\:01\:00.1/sriov_numvfs
+   ```
+
+5. Verify that these were added to the host:
+
+   ```
+   # lspci | grep Ethernet | grep Virtual
+   02:10.0 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+   02:10.1 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+   ```
+
+6. Assign MAC address to each VF
+
+   Depending on the NIC being used, you may need to explicitly set the MAC
+   address for the VF device in order to guarantee that the address is 
+   consistent on the host and when passed through to the guest.
+   For example,
+
+   ```
+   $ ip link set <pf> vf <vfidx> mac <fake MAC address>
+   ```
+
+## Example: Launch a Clear Container using SR-IOV
```

Comment:
Example: Launching a Clear Container Using SR-IOV

The following example launches a Clear Container using SR-IOV: